CONTRIBUTING.md

```diff
@@ -1,89 +1,87 @@
 # Contributing to the Kubeflow SDK
 
-Thank you for your interest in contributing to the Kubeflow SDK!
+This guide explains how to contribute to the Kubeflow SDK project.
+For the Kubeflow SDK documentation, please check [the official Kubeflow documentation](https://www.kubeflow.org/docs/components/).
 
-## Getting Started
-
-### Prerequisites
-- Python 3.8–3.11
-- [pip](https://pip.pypa.io/en/stable/)
+## Requirements
+- [Supported Python version](./pyproject.toml#L4)
 - [pre-commit](https://pre-commit.com/)
+- [uv](https://docs.astral.sh/uv/getting-started/installation/)
 
-### Setting Up Your Development Environment
-Clone the repository:
-```sh
-git clone https://github.com/kubeflow/sdk.git
-cd sdk
-```
 
-Create a virtual environment and activate it:
-```sh
-python3 -m venv .venv
-source .venv/bin/activate
-```
+## Development
 
-Install dependencies in editable mode:
-```sh
-pip install -e .
-```
+The Kubeflow SDK project includes a Makefile with several helpful commands to streamline your development workflow.
+
+To install all dependencies (including dev tools) and create virtual environment, run
 
-#### Development Build (Optional)
-To install development tools and the latest API modules directly from the master branch:
 ```sh
-uv pip install -e . --group dev
+make install-dev
 ```
 
-## Development Workflow
+### Coding Style
+Make sure to install [pre-commit](https://pre-commit.com/) (`uv pip install pre-commit`) and run `pre-commit install` from the root of the repository at least once before creating git commits.
+
+The pre-commit hooks ensure code quality and consistency. They are executed in CI. PRs that fail to comply with the hooks will not be able to pass the corresponding CI gate. The hooks are only executed against staged files unless you run `pre-commit run --all`, in which case, they'll be executed against every file in the repository.
+
+Specific programmatically generated files listed in the `exclude` field in [.pre-commit-config.yaml](.pre-commit-config.yaml) are deliberately excluded from the hooks.
 
-### Pre-commit
+To check formatting:
+
+```shell
+make verify 
+```
+
+#### Pre-commit
 We use pre-commit to ensure consistent code formatting. To enable pre-commit hooks, run:
-```sh
+```shell
 pre-commit install
 ```
 To run all hooks manually:
 ```sh
 pre-commit run --all-files
```

Comment:
You can remove those sections, since you explain it here: https://github.com/kubeflow/sdk/blob/afe4c9160412090c885e9ab5c214d34c6ea788c3/CONTRIBUTING.md#coding-style

```suggestion
```